cmd/cspc-operator/app/handler.go

```diff
@@ -310,25 +316,14 @@ func (pc *PoolConfig) deleteAssociatedCSPI() error {
 }
 
 // removeSPCFinalizer removes CSPC finalizers on associated
-// BDC resources and CSPC object itself.
+// BDC and CSPI resources in correct order and CSPC object itself.
 func (pc *PoolConfig) removeCSPCFinalizer() error {
-	cspList, err := apiscsp.NewKubeClient().List(metav1.ListOptions{
-		LabelSelector: string(apis.StoragePoolClaimCPK) + "=" + pc.AlgorithmConfig.CSPC.Name,
-	})
 
+	// clean up all cspi related resources for given cspc
+	err := cleanupCSPIResources(pc.AlgorithmConfig.CSPC)
 	if err != nil {
-		return errors.Wrap(err, "failed to remove CSPC finalizer on associated resources")
-	}
-
-	if len(cspList.Items) > 0 {
```

Comment:
can we keep this check to make sure that CSPIs related to this CSPC are all really gone